src/aggregations/filter-aggregation.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Construct a filter with script for aggregation.
+ *
+ * @param  {String} filterName Filter Field name for reffernace.
+ * @param  {String} filterRef Field name for filterName to referance in the aggrigation.
+ * @param  {String} operator  Operator for script.
+ * @param  {String} value  Value of script.
+ * @return {Object}       Script Aggregation.
+ */
+export default function filterAggregation(filterName, filterRef, operator, value) {
+  return {
+    [filterName]: {
+      bucket_selector: {
+        buckets_path: {
+          [filterName]: filterRef
+        },
+        script: `${filterName} ${operator} ${value}`
```

Comment:
This is great, but is focused really heavily on using this aggregation with an inline script and bucket_selector, which is not always the desired approach https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-filter-aggregation.html.

I think we can support these more advanced options, but it should be an opt-in custom option. So I'm thinking a function signature like

```
function filterAggregation(filterType, filterField, filterValue)
```

would be ideal here. What do you think?